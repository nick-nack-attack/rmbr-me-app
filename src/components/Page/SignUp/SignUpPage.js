// component for user to create an account
import React, { Component } from 'react';

// component
import SignUpForm from "../../Form/SignUp/SignUpForm";

export default class SignUpPage extends Component {

    static defaultProps = {
        history: {
            push: () => {},
        },
    }

    handleSignUpSuccess = () => {
        window.location.reload(true);
    }

    render() {
        return (
                <SignUpForm
                    onSignUpSuccess={this.handleSignUpSuccess}
                />
        )
    }
}