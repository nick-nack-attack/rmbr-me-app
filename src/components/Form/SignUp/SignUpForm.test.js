import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SignUp from "./SignUp";

describe(`SignUpForm component`, () => {

    it('renders SignUpForm by default', () => {
        const wrapper = shallow(<SignUp/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})