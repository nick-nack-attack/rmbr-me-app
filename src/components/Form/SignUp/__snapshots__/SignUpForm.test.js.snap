// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm component renders SignUpForm by default 1`] = `
<form
  className="SignUpForm"
  onSubmit={[Function]}
>
  <legend>
    <h2>
      Sign Up
    </h2>
  </legend>
  <div
    className="user_name"
  >
    <label
      htmlFor="RegistrationForm__user_name"
    >
      Email 
      <Required />
    </label>
    <Input
      id="RegistrationForm__user_name"
      name="user_name"
      onChange={[Function]}
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm__password"
    >
      Password 
      <Required />
    </label>
    <Input
      id="RegistrationForm__password"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
    />
  </div>
  <Button
    type="submit"
  >
    Get started!
  </Button>
  <div
    role="alert"
  />
</form>
`;
